sway/input/cursor.c

```diff
@@ -179,6 +181,8 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 			if (!do_mouse_focus) {
 				struct sway_container *next_focus = seat_get_focus_inactive(
 						cursor->seat, p);
+				sway_assert(next_focus->type == C_VIEW,
+							"focus inactive container is not a view");
```

Comment:
`next_focus` may be `NULL` which would make this crash.

Also, when sway encounters a failing `sway_assert` in a **development** build, it'll crash with the message given. When sway encounters a failing `sway_assert` in a **release** build, the `sway_assert` will return false and it'll continue on executing code. Which in your case would lead to a crash. This is why other places in the code use a pattern of:

    if (!sway_assert(condition, "message")) {
        return;
    }